python/tvm/driver/tvmc/autotuner.py

```diff
@@ -453,7 +518,37 @@ def tune_model(
             hardware_params=hardware_params,
             include_simple_tasks=include_simple_tasks,
         )
+    else:
+        tasks = autotvm_get_tuning_tasks(
+            mod=mod,
+            params=params,
+            target=target,
+            alter_layout=desired_layout,
+        )
+
+    # Filter extracted tasks by provided user expression
+    if tasks_filter:
+        tasks, do_list = filter_tasks(tasks, tasks_filter)
+        if do_list:
+            print("Available Tasks for tuning:")
+            print(
+                "\n".join(
+                    [
+                        "  {}. {}".format(
+                            i, task if len(str(task)) < 100 else str(task)[:97] + "..."
```

Comment:
in this case does this make the task useless?

Reply:
Sorry, but I do not get the question… This truncates the string representation of each task to 100 characters which is typically enough for AutoTVM task to show the operator name and input shape.

Or do you mean what would happen in a situation where `list` as well as a filter expression is passed by the user?

Reply:
oh sorry, i meant to clarify this point before sending and then i forgot to delete it after realizing it was moot. originally i read the diff before the full description and thought that this was more like a filter-by-name command-line option. so, please ignore :)